Terminal.Gui/Views/DatePicker.cs

```diff
@@ -0,0 +1,298 @@
+
+/// <summary>
+/// The <see cref="DatePicker"/> <see cref="View"/> Date Picker.
+/// </summary>
+public class DatePicker : TextField {
+
+	private DataTable table;
+	private TableView calendar;
+	private DateTime date = DateTime.Now;
+	private List<string> months = new ();
+	private List<string> years = new ();
+
+	/// <summary>
+	/// Format of date. The default is MM/dd/yyyy.
+	/// </summary>
+	public string Format { get; set; } = CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern;
+
+	/// <summary>
+	/// Range of years in the calendar. The default is 1900..2100.
+	/// </summary>
+	public Range YearsRange { get; set; } = 1900..2100;
+
+	/// <summary>
+	/// Get or set the date.
+	/// </summary>
+	public DateTime Date {
+		get => date;
+		set {
+			date = value;
+			Text = date.ToString (Format);
+		}
+	}
+
+	/// <summary>
+	/// Initializes a new instance of <see cref="DatePicker"/>.
+	/// </summary>
+	public DatePicker () : base () => Initialize (date);
+
+	/// <summary>
+	/// Initializes a new instance of <see cref="DatePicker"/> with the specified date.
+	/// </summary>
+	public DatePicker (DateTime date) : base ()
+	{
+		Initialize (date);
+	}
+
+	/// <summary>
+	/// Initializes a new instance of <see cref="DatePicker"/> with the specified date and format.
+	/// </summary>
+	public DatePicker (DateTime date, string format) : base ()
+	{
+		Format = format;
+		Initialize (date);
+	}
+
+	private void Initialize (DateTime date)
+	{
+		Date = date;
+		months = Enum.GetValues (typeof (Month)).Cast<Month> ().Select (x => x.ToString ()).ToList ();
+	}
+
+	private void ShowDatePickerDialog ()
+	{
+		years = Enumerable.Range (YearsRange.Start.Value, YearsRange.End.Value - YearsRange.Start.Value + 1)
+			.Select (x => x.ToString ())
+			.ToList ();
+
+		if (date.Year < YearsRange.Start.Value || date.Year > YearsRange.End.Value) {
+			MessageBox.ErrorQuery ("Error", "Date year is out of range", "Ok");
+			return;
+		}
+
+		var dialog = new Dialog () {
+			Height = 14,
+			Width = 40,
+			Title = "Date Picker",
+		};
+
+		calendar = new TableView () {
+			X = 0,
+			Y = Pos.Center () + 1,
+			Width = 22,
+			Height = Dim.Fill (1),
+			Style = new TableStyle {
+				ShowHeaders = true,
+				ShowHorizontalHeaderOverline = true,
+				ShowHorizontalHeaderUnderline = true,
+				ShowHorizontalBottomline = true,
+				ShowVerticalCellLines = true,
+				ExpandLastColumn = false,
+			}
+		};
+
+		var yearsLabel = new Label ("Year:") {
+			X = Pos.Right (calendar) + 1,
+			Y = 1,
+			Height = 1,
+		};
+		var comboBoxYear = new ComboBox (years) {
+			X = Pos.Right (calendar) + 1,
+			Y = 2,
+			Width = 12,
+			Height = 4,
+			SelectedItem = years.IndexOf (date.Year.ToString ())
+		};
+		comboBoxYear.SelectedItemChanged += ChangeYearDate;
+
+		var monthsLabel = new Label ("Month:") {
+			X = Pos.Right (calendar) + 1,
+			Y = 6,
+			Height = 1,
+		};
+
+		var comboBoxMonth = new ComboBox (months) {
+			X = Pos.Right (calendar) + 1,
+			Y = 7,
+			Width = 12,
```

Comment:
Ditto.

Reply:
(not gonna comment any more on the rest of the int literals. You get the point).

Reply:
Done, yeah I get the point 😁